src/main/java/de/tum/in/www1/artemis/repository/ExamRepository.java

```diff
@@ -53,11 +53,22 @@
     Optional<Exam> findWithStudentExamsExercisesById(Long id);
 
     @Query("""
-            select e
-            from Exam e, User u
-            where u.id = :#{#userId} and e.course.instructorGroupName member of u.groups
+            select distinct e
+            from Exam e left join fetch e.exerciseGroups eg left join fetch eg.exercises ex, User u
+            where u.id = :#{#userId} and e.course.instructorGroupName member of u.groups and
+            exists (select exercise
+                from Exercise exercise
+                where exercise.exerciseGroup.exam.id = e.id and TYPE(exercise) = QuizExercise)
```

Comment:
I think this can be done with join, avoiding the nested query.

Reply:
👍